validator/src/main.rs

```diff
@@ -1529,8 +1529,16 @@ pub fn main() {
         packager_thread_niceness_adj: snapshot_packager_niceness_adj,
     };
 
+    // The accounts hash interval shall match the snapshot interval, unless manually specified
     validator_config.accounts_hash_interval_slots =
-        value_t_or_exit!(matches, "accounts-hash-interval-slots", u64);
+        match matches.value_of("accounts_hash_interval_slots") {
+            Some(_) => value_t_or_exit!(matches, "accounts_hash_interval_slots", Slot),
+            None => std::cmp::min(
+                full_snapshot_archive_interval_slots,
+                incremental_snapshot_archive_interval_slots,
```

Comment:
is a full interval less than the incremental interval actually a legal configuration?

Reply:
heh, `0` would've been a more intuitive "disabled" sentinel, assuming `Option<Slot>` wasn't viable for some reason

Reply:
you'll never guess how rust represents `Option::<NonZero*>::None` under the hood :wink: